lib/Partitioner/Partitioner.cpp

```diff
@@ -492,7 +501,78 @@ NodeToFunctionMap Partitioner::selectPartitions(Function *F,
 /// the number of DAGNodes) is larger than the number of devices. E.g:
 /// node1(6GB) -> node2(14GB) -> node3(6GB). The memory limitation is 16GB, and
 /// there is only 2 devices.
-void Partitioner::adjustLogicalDeviceID() {}
+llvm::Error Partitioner::adjustLogicalDeviceID(NodeToFunctionMap &mapping) {
+  for (auto &dag : partitions_) {
+    // Traver the DAGNodes, and get list of with each backendKind.
+    std::map<BackendKind, std::vector<DAGNode *>> backendKindNum;
```

Comment:
The name `backendKindNum` is a bit odd as you do not map the backendKind to a number. If I understand correctly, you map the backendKind to a set of DAGNodes and then you check the number of elements in this set.